src/tasks/minions/minigames/temporossActivity.ts

```diff
@@ -0,0 +1,104 @@
+import { increaseNumByPercent } from 'e';
+import { Task } from 'klasa';
+
+import { Emoji, Events } from '../../../lib/constants';
+import { getTemporossLoot } from '../../../lib/simulation/tempoross';
+import Fishing from '../../../lib/skilling/skills/fishing';
+import { SkillsEnum } from '../../../lib/skilling/types';
+import { ItemBank } from '../../../lib/types';
+import { TemporossActivityTaskOptions } from '../../../lib/types/minions';
+import { bankHasItem, channelIsSendable } from '../../../lib/util';
+import { handleTripFinish } from '../../../lib/util/handleTripFinish';
+import itemID from '../../../lib/util/itemID';
+
+export default class extends Task {
+	async run(data: TemporossActivityTaskOptions) {
+		const { userID, channelID, quantity, rewardBoost, duration } = data;
+		const user = await this.client.users.fetch(userID);
+		const currentLevel = user.skillLevel(SkillsEnum.Fishing);
+		const channel = await this.client.channels.fetch(channelID);
+
+		let loot: ItemBank = {};
```

Comment:
this is coded weirdly: you define an empty ItemBank, then re-assign it below ,then you use the less-preferred `bankHasItem` function.

You should simply do: `const loot = getTemporossLoot()`, and then instead of `bankHasItem(loot, itemID('Tiny tempor'))`, just do `loot.has('Tiny tempor')` You shouldnt be using `ItemBank` anywhere anymore, unless its a special circumstance.